src/h5web/vis-packs/core/shared/Html.tsx

```diff
@@ -32,36 +33,50 @@ const Html = forwardRef<HTMLDivElement, HtmlProps>((props, ref) => {
   } = props;
 
   const { width, height } = useThree((state) => state.size);
+  const camera = useThree((state) => state.camera);
   const gl = useThree((state) => state.gl);
   const { parentElement } = gl.domElement;
-  const camera = useThree((state) => state.camera);
 
   // Container `div` for ReactDOM to render into, appended next to R3F's `canvas`
-  const [el] = useState(() => document.createElement('div'));
-  // Placeholder R3F group, the position of which is tracked and forwarded to `el` when `followCamera` is enabled
+  const [el] = useState(() => {
+    const elem = document.createElement('div');
+
+    // Take out of flow and let pointer events pass through to canvas
+    elem.style.cssText = `position: absolute; top: 0; left: 0; pointer-events: none;`;
+
+    return elem;
+  });
+
+  // Inner `div` that is transformed and scaled if `followCamera` and `scaleOnZoom` are enabled
+  const innerRef = useRef<HTMLDivElement>(null);
+
+  /* Placeholder R3F group, the position of which is tracked and forwarded to
+   * the inner `div` if `followCamera` and `scaleOnZoom` are enabled. */
   const group = useRef<Group>(null!); // eslint-disable-line @typescript-eslint/no-non-null-assertion
 
-  function updateHtmlPosition() {
-    const objectPos = v1.setFromMatrixPosition(group.current.matrixWorld);
-    objectPos.project(camera);
-    const widthHalf = width / 2;
-    const heightHalf = height / 2;
-    const pos = [
-      objectPos.x * widthHalf + widthHalf,
-      -(objectPos.y * heightHalf) + heightHalf,
+  const getGroupPosition = useCallback(() => {
+    const groupPos = v1.setFromMatrixPosition(group.current.matrixWorld);
+    groupPos.project(camera);
+
+    return [
+      groupPos.x * (width / 2) + width / 2,
+      -(groupPos.y * (height / 2)) + height / 2,
     ];
+  }, [camera, height, width]);
 
-    el.style.transform = `translate3d(${pos[0]}px,${pos[1]}px,0)
-    scale(${scaleOnZoom ? camera.zoom : 1})`;
-  }
+  const getInnerDivTransform = useCallback(() => {
+    const position = followCamera ? getGroupPosition() : undefined;
 
-  useLayoutEffect(() => {
-    group.current.updateMatrixWorld(); // To place correctly `el` on first render
-    el.style.cssText = `position:absolute; top:0; left:0; transform-origin:0 0;`;
-    if (followCamera) {
-      updateHtmlPosition();
-    }
+    return [
+      position && `translate3d(${position[0]}px, ${position[1]}px, 0)`,
+      scaleOnZoom ? `scale(${camera.zoom})` : undefined,
+    ]
+      .filter(isDefined)
+      .join(' ');
+  }, [camera, followCamera, getGroupPosition, scaleOnZoom]);
```

Comment:
This `useCallback` returns the `transform` string, which is empty if both `scaleOnZoom` and `followCamera` are disabled.